modules/core/client/components/Map/index.js

```diff
@@ -38,9 +38,10 @@ export default function Map(props) {
       mapStyle={MAP_STYLE_DEFAULT}
       onViewportChange={setViewport}
       touchRotate={false}
-      width="100%"
-      {...props}
```

Comment:
By removing `props`, don't we lose the ability to reuse the `Map` component in a flexible way?
e.g. we may want to set the `Map`'s width and height, or control the state from outside (I'd like to do both of these for refactoring offer-host-edit).

Wouldn't something like the following work? (I haven't figured out details.)

```jsx
{...defaults}
{...props} /* provided props overwrite defaults */
{...viewport} /* zoom, latitude and longitude overwrite the props */
/* only zoom, latitude and longitude get updated */
onViewportChange={({ zoom, latitude, longitude }) => setViewport({ zoom, latitude, longitude })}
```

But I suspect this won't allow me to tap into the internal zoom, latitude and longitude state anyways, since the `viewport` overwrites whatever we provide. Probably fit for a different PR.

Reply:
Ah I see, yes, we can reintroduce props, but important thing is to only pass down the props that we don't use directly... I can implement that now :)

Reply:
The `onViewportChange` callback has loads of extra stuff it sets beyond just zoom, latitude, longitude, so I quite liked keeping it intact.

See https://github.com/Trustroots/trustroots/pull/1332/commits/8936d4095e1dd44d90d222c884a765ac43ecf6e6 for passing down extra props. Although I guess we'd get a lint error if we don't declare what we want to pass down in the prop types...